### PR TITLE
PN-7404-fe-rework-test-generale-pf

### PR DESCRIPTION
**Pull request by ImJimMoriarty**
## Short description
Adjusted mocks and sorted all api mocks inside `__mocks__`.

## List of changes proposed in this pull request
- Added to each test:

> - `beforeAll(() => {
    mock = new MockAdapter(apiClient);
  });` to instantiate the mock;
> - `afterEach(() => {
    mock.reset();
  });` to reset the mock each test;
> - `afterAll(() => {
    mock.restore();
  });` to restore the mock once all test are made;

- Added directory `__mocks__`  and move all `test-utils` inside.
- Removed `MockAdapter` instances.
- Sorted and and adjusted the code.

## How to test
Just run tests into `packages\pn-personafisica-webapp\src\api`

### CODE REVIEW

File: packages/pn-personafisica-webapp/src/api/notifications/__test__/Notifications.api.test.ts [hunk 1]
```diff
@@ -36,45 +36,77 @@ describe('Notifications api tests', () => {
...
       startDate: formatToTimezoneString(tenYearsAgo),
       endDate: formatToTimezoneString(getNextDay(today)),
     });
     expect(res).toStrictEqual(notificationsToFe);
-    mock.reset();
-    mock.restore();
+    cleanupMock(mock);
+  });
+
+  it('getReceivedNotifications with invalid date range', async () => {
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/api/notifications/__test__/Notifications.api.test.ts:51]: as before

File: packages/pn-personafisica-webapp/src/api/notifications/__test__/Notifications.api.test.ts [hunk 2]
```diff
@@ -99,28 +133,34 @@ describe('Notifications api tests', () => {
...
       category: LegalFactType.ANALOG_DELIVERY,
     };
-    const mock = new MockAdapter(apiClient);
-    mock.onGet(NOTIFICATION_DETAIL_LEGALFACT(iun, legalFact)).reply(200, undefined);
+    const mock = mockApi(
+      apiClient,
+      'GET',
+      NOTIFICATION_DETAIL_LEGALFACT(iun, legalFact),
+      200,
+      undefined,
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/api/notifications/__test__/Notifications.api.test.ts:141]: you can omit the two undefined

File: packages/pn-personafisica-webapp/src/api/notifications/__test__/Notifications.api.test.ts [hunk 3]
```diff
@@ -177,25 +220,37 @@ describe('Notifications api tests', () => {
...
+      'POST',
+      NOTIFICATION_ID_FROM_QRCODE(),
+      200,
+      { aarQrCodeValue: 'qr1' },
+      {
+        iun: 'mock-notification-1',
+        mandateId: 'mock-mandate-1',
+      }
+    );
     mock.onPost(NOTIFICATION_ID_FROM_QRCODE(), { aarQrCodeValue: 'qr2' }).reply(200, {
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/api/notifications/__test__/Notifications.api.test.ts:238]: the mock of qr2 is useless. you can remove it

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-personafisica-webapp/src/__test__/test-utils.tsx [hunk 4]
```diff
@@ -85,10 +85,20 @@ function mockApi(
   return mock;
 }
 
+/**
+ * Utility function to cleanup a Mock
+ * @param client Axios client or Mock Adapter instance
+ */
+function cleanupMock(client: AxiosInstance | MockAdapter) {
```
[the remark this answers is not in the document]
Comment (ImJimMoriarty) [reply packages/pn-personafisica-webapp/src/__test__/test-utils.tsx]: changes made into 1bf7836db3775121009b8ac49703ef1bd0689074

File: packages/pn-personafisica-webapp/src/api/consents/__test__/Consents.api.test.ts [hunk 5]
```diff
@@ -5,25 +5,68 @@ import { ConsentActionType, ConsentType } from '../../../models/consents';
...
+  });
+
+  it('setConsentByType with status code 400', async () => {
+    const mock = mockApi(
+      apiClient,
+      'PUT',
+      SET_CONSENTS(ConsentType.TOS, 'mocked-version-1'),
+      400,
+      undefined,
+      { error: 'Invalid request' },
```
[the remark this answers is not in the document]
Comment (ImJimMoriarty) [reply packages/pn-personafisica-webapp/src/api/consents/__test__/Consents.api.test.ts]: changes made into 1bf7836db3775121009b8ac49703ef1bd0689074

File: packages/pn-personafisica-webapp/src/api/delegations/__test__/Delegations.api.test.ts [hunk 6]
```diff
@@ -58,6 +53,12 @@ describe('Delegations api tests', () => {
     expect(res).toHaveLength(0);
   });
 
+  it('gets delegates with status code 400', async () => {
```
[the remark this answers is not in the document]
Comment (ImJimMoriarty) [reply packages/pn-personafisica-webapp/src/api/delegations/__test__/Delegations.api.test.ts]: changes made into 1bf7836db3775121009b8ac49703ef1bd0689074

File: packages/pn-personafisica-webapp/src/api/notifications/__test__/Notifications.api.test.ts (hunk 3 above)
Comment (ImJimMoriarty) [reply packages/pn-personafisica-webapp/src/api/notifications/__test__/Notifications.api.test.ts]: changes made into 1bf7836db3775121009b8ac49703ef1bd0689074

File: packages/pn-personafisica-webapp/src/api/consents/__test__/Consents.api.test.ts [hunk 7]
```diff
@@ -5,25 +5,68 @@ import { ConsentActionType, ConsentType } from '../../../models/consents';
...
+    );
+    await expect(
```
[the remark this answers is not in the document]
Comment (ImJimMoriarty) [reply packages/pn-personafisica-webapp/src/api/consents/__test__/Consents.api.test.ts]: changes made into 1bf7836db3775121009b8ac49703ef1bd0689074

File: packages/pn-personafisica-webapp/src/api/delegations/__test__/Delegations.api.test.ts [hunk 8]
```diff
@@ -17,36 +17,35 @@ import {
   REJECT_DELEGATION,
   REVOKE_DELEGATION,
 } from '../delegations.routes';
+import { mockApi } from '../../../__test__/test-utils';
 
 async function getDelegates(response: Array<Delegation> | null) {
-  const axiosMock = new MockAdapter(apiClient);
-  axiosMock.onGet(DELEGATIONS_BY_DELEGATOR()).reply(200, response);
+  mockApi(apiClient, 'GET', DELEGATIONS_BY_DELEGATOR(), 200, undefined, response);
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/api/delegations/__test__/Delegations.api.test.ts:23]: Here you are not setting mock instance and so it isn't neither restored or resetted

File: packages/pn-personafisica-webapp/src/api/delegations/__test__/Delegations.api.test.ts [hunk 9]
```diff
@@ -69,55 +68,39 @@ describe('Delegations api tests', () => {
   });
 
   it('revokes a delegation', async () => {
-    const mock = new MockAdapter(apiClient);
-    mock.onPatch(REVOKE_DELEGATION('7')).reply(204);
+    mockApi(apiClient, 'PATCH', REVOKE_DELEGATION('7'), 204, undefined);
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/api/delegations/__test__/Delegations.api.test.ts:71]: as before

File: packages/pn-personafisica-webapp/src/api/delegations/__test__/Delegations.api.test.ts [hunk 10]
```diff
@@ -69,55 +68,39 @@ describe('Delegations api tests', () => {
...
     const res = await DelegationsApi.revokeDelegation('7');
     expect(res).toStrictEqual({ id: '7' });
-    mock.reset();
-    mock.restore();
   });
 
   it("doesn't revoke a delegation", async () => {
-    const mock = new MockAdapter(apiClient);
-    mock.onPatch(REVOKE_DELEGATION('10')).reply(200);
+    mockApi(apiClient, 'PATCH', REVOKE_DELEGATION('10'), 200, undefined);
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/api/delegations/__test__/Delegations.api.test.ts:77]: remove useless undefined here and after

File: packages/pn-personafisica-webapp/src/api/external-registries/__test__/External-registries.api.test.ts [hunk 11]
```diff
@@ -1,18 +1,24 @@
...
-    mock.onGet(GET_ALL_ACTIVATED_PARTIES(undefined)).reply(200, []);
+  afterEach(() => {
+    if (mock) {
+      mock.restore();
+      mock.reset();
+    }
+  });
+
+  it('getAllActivatedParties 200', async () => {
+    mockApi(apiClient, 'GET', GET_ALL_ACTIVATED_PARTIES(undefined), 200, undefined, []);
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/api/external-registries/__test__/External-registries.api.test.ts:20]: as before you are not setting mock istance. check all tests

File: packages/pn-personafisica-webapp/src/api/notifications/__test__/Notifications.api.test.ts [hunk 12]
```diff
@@ -31,50 +30,70 @@ import {
...
     const res = await NotificationsApi.getReceivedNotifications({
       startDate: formatToTimezoneString(tenYearsAgo),
       endDate: formatToTimezoneString(getNextDay(today)),
     });
     expect(res).toStrictEqual(notificationsToFe);
-    mock.reset();
-    mock.restore();
+  });
+
+  it('getReceivedNotifications with server error', async () => {
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/api/notifications/__test__/Notifications.api.test.ts:57]: error check is useless

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]
Also the order of the imports must be fixed. use a vs extension to do this

File: packages/pn-personafisica-webapp/src/api/notifications/__test__/Notifications.api.test.ts (hunk 12 above)
Comment (ImJimMoriarty) [reply packages/pn-personafisica-webapp/src/api/notifications/__test__/Notifications.api.test.ts]: changes made into: [430db73](https://github.com/pagopa/pn-frontend/pull/938/commits/430db73dd4e6ee0215846db2b30db4717bdd1763)

File: packages/pn-personafisica-webapp/src/api/external-registries/__test__/External-registries.api.test.ts (hunk 11 above)
Comment (ImJimMoriarty) [reply packages/pn-personafisica-webapp/src/api/external-registries/__test__/External-registries.api.test.ts]: changes made into: [430db73](https://github.com/pagopa/pn-frontend/pull/938/commits/430db73dd4e6ee0215846db2b30db4717bdd1763)

File: packages/pn-personafisica-webapp/src/api/delegations/__test__/Delegations.api.test.ts (hunk 10 above)
Comment (ImJimMoriarty) [reply packages/pn-personafisica-webapp/src/api/delegations/__test__/Delegations.api.test.ts]: changes made into: [430db73](https://github.com/pagopa/pn-frontend/pull/938/commits/430db73dd4e6ee0215846db2b30db4717bdd1763)

File: packages/pn-personafisica-webapp/src/api/delegations/__test__/Delegations.api.test.ts (hunk 9 above)
Comment (ImJimMoriarty) [reply packages/pn-personafisica-webapp/src/api/delegations/__test__/Delegations.api.test.ts]: changes made into: [430db73](https://github.com/pagopa/pn-frontend/pull/938/commits/430db73dd4e6ee0215846db2b30db4717bdd1763)

File: packages/pn-personafisica-webapp/src/api/delegations/__test__/Delegations.api.test.ts (hunk 8 above)
Comment (ImJimMoriarty) [reply packages/pn-personafisica-webapp/src/api/delegations/__test__/Delegations.api.test.ts]: changes made into: [430db73](https://github.com/pagopa/pn-frontend/pull/938/commits/430db73dd4e6ee0215846db2b30db4717bdd1763)

File: packages/pn-personafisica-webapp/src/component/Contacts/__test__/CourtesyContactItem.test.tsx [hunk 13]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-
+import MockAdapter from 'axios-mock-adapter';
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/component/Contacts/__test__/CourtesyContactItem.test.tsx:2]: if you define let mock, in each you must delete const mock and leave only mock.
The afterEach method runs code efter each test, so the mock.restore and mock.reset must be deleted from the single test. 
Please check all tests

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-personafisica-webapp/src/redux/dashboard/__test__/reducers.test.ts [hunk 14]
```diff
@@ -6,16 +7,26 @@ import {
...
+import { mockApi } from '../../../__test__/test-utils';
+import { apiClient } from '../../../api/apiClients';
 import { NotificationsApi } from '../../../api/notifications/Notifications.api';
 import { mockAuthentication } from '../../auth/__test__/test-utils';
 import { store } from '../../store';
 import { getReceivedNotifications } from '../actions';
 import { setMandateId, setNotificationFilters, setPagination, setSorting } from '../reducers';
-import { notificationsToFe } from './test-utils';
+import { notificationsFromBe, notificationsToFe } from './test-utils';
+import { NOTIFICATIONS_LIST } from '../../../api/notifications/notifications.routes';
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/redux/dashboard/__test__/reducers.test.ts:18]: Order imports. ../../../ goes before ../../ and so on ../

File: packages/pn-personafisica-webapp/src/redux/newDelegation/__test__/reducers.test.ts [hunk 15]
```diff
@@ -10,25 +12,45 @@ import {
   createDelegationSelectedPayload,
   initialState,
 } from './test.utils';
+import { mockApi } from '../../../__test__/test-utils';
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/redux/newDelegation/__test__/reducers.test.ts:15]: Order imports as before

File: packages/pn-personafisica-webapp/src/redux/notification/__test__/reducers.test.ts [hunk 16]
```diff
@@ -22,7 +23,10 @@ import {
   getReceivedNotificationOtherDocument,
 } from '../actions';
 import { resetLegalFactState, resetState } from '../reducers';
-import { notificationToFe } from './test-utils';
+import { notificationFromBe, notificationToFe } from './test-utils';
+import { NOTIFICATION_DETAIL, NOTIFICATION_DETAIL_DOCUMENTS, NOTIFICATION_DETAIL_LEGALFACT, NOTIFICATION_DETAIL_OTHER_DOCUMENTS, NOTIFICATION_PAYMENT_ATTACHMENT, NOTIFICATION_PAYMENT_INFO, NOTIFICATION_PAYMENT_URL } from '../../../api/notifications/notifications.routes';
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/redux/notification/__test__/reducers.test.ts:27]: Order imports

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-personafisica-webapp/src/redux/notification/__test__/reducers.test.ts (hunk 16 above)
Comment (ImJimMoriarty) [reply packages/pn-personafisica-webapp/src/redux/notification/__test__/reducers.test.ts]: fixed: https://github.com/pagopa/pn-frontend/pull/938/commits/40331b2cd8c5f4a399c800ec71a77ef4f58cf51e

File: packages/pn-personafisica-webapp/src/redux/newDelegation/__test__/reducers.test.ts (hunk 15 above)
Comment (ImJimMoriarty) [reply packages/pn-personafisica-webapp/src/redux/newDelegation/__test__/reducers.test.ts]: fixed: https://github.com/pagopa/pn-frontend/pull/938/commits/40331b2cd8c5f4a399c800ec71a77ef4f58cf51e

File: packages/pn-personafisica-webapp/src/redux/dashboard/__test__/reducers.test.ts (hunk 14 above)
Comment (ImJimMoriarty) [reply packages/pn-personafisica-webapp/src/redux/dashboard/__test__/reducers.test.ts]: fixed: https://github.com/pagopa/pn-frontend/pull/938/commits/40331b2cd8c5f4a399c800ec71a77ef4f58cf51e

File: packages/pn-personafisica-webapp/src/component/Deleghe/__test__/MobileDelegators.test.tsx [hunk 17]
```diff
@@ -17,7 +16,7 @@ jest.mock('react-i18next', () => ({
 /**
  * Vedi commenti nella definizione di simpleMockForApiErrorWrapper
  */
- jest.mock('@pagopa-pn/pn-commons', () => {
+jest.mock('@pagopa-pn/pn-commons', () => {
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/component/Deleghe/__test__/MobileDelegators.test.tsx:19]: same as before

File: packages/pn-personafisica-webapp/src/pages/__test__/NotificationDetail.page.test.tsx [hunk 18]
```diff
@@ -30,8 +28,8 @@ let mockUseSimpleBreadcrumb = false;
 jest.mock('react-i18next', () => ({
   // this mock makes sure any components using the translate hook can use it without a warning being shown
   useTranslation: () => ({
-      t: (str: string) => str,
-    }),
+    t: (str: string) => str,
+  }),
 }));
 
 jest.mock('react-router-dom', () => {
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/pages/__test__/NotificationDetail.page.test.tsx:35]: remove the mock of useParams and try to mock the window location object.
remove all mocks of all components

File: packages/pn-personafisica-webapp/src/pages/__test__/Notifiche.page.test-utils.tsx [hunk 19]
```diff
@@ -1,12 +1,11 @@
-import React from 'react';
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/pages/__test__/Notifiche.page.test-utils.tsx:1]: Evaluate if move this into the mock of NotificationDetail

File: packages/pn-personafisica-webapp/src/pages/__test__/ToSAcceptance.page.test.tsx [hunk 20]
```diff
@@ -24,6 +26,15 @@ jest.mock('react-router-dom', () => ({
 }));
 
 describe('test Terms of Service page', () => {
+  let mock: MockAdapter;
+
+  afterEach(() => {
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/pages/__test__/ToSAcceptance.page.test.tsx:31]: use the same syntax in the guide

File: packages/pn-personafisica-webapp/src/redux/contact/__test__/reducers.test.ts [hunk 21]
```diff
@@ -36,26 +37,31 @@ const initialState = {
 };
 
 describe('Contacts redux state tests', () => {
+  let mock: MockAdapter;
   mockAuthentication();
+  afterEach(() => {
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/redux/contact/__test__/reducers.test.ts:42]: same as before

File: packages/pn-personafisica-webapp/src/redux/dashboard/__test__/reducers.test.ts [hunk 22]
```diff
@@ -1,21 +1,30 @@
...
 import { store } from '../../store';
 import { getReceivedNotifications } from '../actions';
 import { setMandateId, setNotificationFilters, setPagination, setSorting } from '../reducers';
-import { notificationsToFe } from './test-utils';
 
 describe('Dashbaord redux state tests', () => {
+  let mock: MockAdapter;
   mockAuthentication();
+  afterEach(() => {
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/redux/dashboard/__test__/reducers.test.ts:22]: same as before. check all tests

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-personafisica-webapp/src/__test__/App.test.tsx [hunk 23]
```diff
@@ -1,26 +1,24 @@
...
-import { Suspense } from 'react';
-import * as redux from 'react-redux';
-
-/* eslint-disable import/order */
-import { axe, render } from './test-utils';
 import App from '../App';
 import i18n from '../i18n';
 import * as sidemenuActions from '../redux/sidemenu/actions';
 
+/* eslint-disable import/order */
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/__test__/App.test.tsx:6]: remove this comment

File: packages/pn-personafisica-webapp/src/__test__/App.test.tsx [hunk 24]
```diff
@@ -93,11 +90,10 @@ describe('App', () => {
     jest.restoreAllMocks();
   });
 
-
   /**
    * Tests che usano Component e inizializzazione "semplice" di i18n.
    */
-  describe("tests che non analizzano dettagli (test solo di renderizzazione)", () => {
+  describe('tests che non analizzano dettagli (test solo di renderizzazione)', () => {
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/__test__/App.test.tsx:96]: maybe we have to rework these tests. check the pa, paying attention to different behaviour between the two apps

File: packages/pn-personafisica-webapp/src/component/Contacts/__test__/CourtesyContactsList.a11y.test.tsx [hunk 25]
```diff
@@ -1,9 +1,9 @@
-import * as React from 'react';
-import { act, RenderResult } from "@testing-library/react";
 import { fail } from 'assert';
+import * as React from 'react';
+import { act, RenderResult } from "../../../__test__/test-utils";
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/component/Contacts/__test__/CourtesyContactsList.a11y.test.tsx:3]: Merge this import with the next one

File: packages/pn-personafisica-webapp/src/component/Contacts/__test__/CourtesyContactsList.test.tsx [hunk 26]
```diff
@@ -1,11 +1,11 @@
...
-import { act, RenderResult, screen } from '@testing-library/react';
 import * as redux from 'react-redux';
+import { act, render, screen } from '../../../__test__/test-utils';
 import { CourtesyChannelType, DigitalAddress } from '../../../models/contacts';
 import * as hooks from '../../../redux/hooks';
-import { render } from '../../../__test__/test-utils';
 import CourtesyContactsList from '../CourtesyContactsList';
 import { DigitalContactsCodeVerificationProvider } from '../DigitalContactsCodeVerification.context';
+
 // import * as actions from '../../../redux/contact/actions';
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/component/Contacts/__test__/CourtesyContactsList.test.tsx:9]: remove commented code

File: packages/pn-personafisica-webapp/src/component/Contacts/__test__/DigitalContactElem.test.tsx [hunk 27]
```diff
@@ -1,17 +1,17 @@
+import { TextField } from '@mui/material';
 import React from 'react';
-import { fireEvent, RenderResult, waitFor, screen } from '@testing-library/react';
 import * as redux from 'react-redux';
-import { TextField } from '@mui/material';
+import { RenderResult, screen, waitFor } from '../../../__test__/test-utils';
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/component/Contacts/__test__/DigitalContactElem.test.tsx:4]: merge this import with that at row 8

File: packages/pn-personafisica-webapp/src/component/Contacts/__test__/DigitalContactsCodeVerification.context.test.tsx [hunk 28]
```diff
@@ -1,11 +1,11 @@
 import * as React from 'react';
-import { fireEvent, RenderResult, screen, waitFor } from '@testing-library/react';
 import * as redux from 'react-redux';
+import { fireEvent, RenderResult, screen, waitFor } from '../../../__test__/test-utils';
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/component/Contacts/__test__/DigitalContactsCodeVerification.context.test.tsx:3]: merge this import with the next one

File: packages/pn-personafisica-webapp/src/component/Contacts/__test__/IOContact.test.tsx [hunk 29]
```diff
@@ -1,13 +1,12 @@
 /* eslint-disable functional/no-let */
 import React from 'react';
-import { fireEvent, RenderResult } from '@testing-library/react';
 import * as redux from 'react-redux';
-import { render, screen } from '../../../__test__/test-utils';
+import { fireEvent, render, RenderResult, screen } from '../../../__test__/test-utils';
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/component/Contacts/__test__/IOContact.test.tsx:4]: merge this import with next one

File: packages/pn-personafisica-webapp/src/component/Contacts/__test__/LegalContactsList.a11y.test.tsx [hunk 30]
```diff
@@ -1,8 +1,8 @@
 import * as React from 'react';
-import { act, RenderResult } from '@testing-library/react';
+import { act, RenderResult } from '../../../__test__/test-utils';
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/component/Contacts/__test__/LegalContactsList.a11y.test.tsx:2]: as before

File: packages/pn-personafisica-webapp/src/component/Contacts/__test__/LegalContactsList.test.tsx [hunk 31]
```diff
@@ -1,12 +1,12 @@
 import React from 'react';
 import * as redux from 'react-redux';
-import { act, fireEvent, RenderResult, waitFor, screen } from '@testing-library/react';
+import { RenderResult, act, fireEvent, screen, waitFor } from '../../../__test__/test-utils';
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/component/Contacts/__test__/LegalContactsList.test.tsx:3]: as before

File: packages/pn-personafisica-webapp/src/component/Contacts/__test__/SpecialContactElem.a11y.test.tsx [hunk 32]
```diff
@@ -1,6 +1,6 @@
-import * as React from 'react';
-import { act, RenderResult } from '@testing-library/react';
 import { SpecialContactsProvider } from '@pagopa-pn/pn-commons';
+import * as React from 'react';
+import { act, RenderResult } from '../../../__test__/test-utils';
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/component/Contacts/__test__/SpecialContactElem.a11y.test.tsx:3]: as before

File: packages/pn-personafisica-webapp/src/component/Contacts/__test__/SpecialContactElem.test.tsx [hunk 33]
```diff
@@ -1,6 +1,6 @@
-import React from 'react';
-import { act, fireEvent, RenderResult, waitFor } from '@testing-library/react';
 import { SpecialContactsProvider } from '@pagopa-pn/pn-commons';
+import React from 'react';
+import { act, fireEvent, RenderResult, waitFor } from '../../../__test__/test-utils';
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/component/Contacts/__test__/SpecialContactElem.test.tsx:3]: as before

File: packages/pn-personafisica-webapp/src/component/Notifications/__test__/FilterNotifications.a11y.test.tsx [hunk 34]
```diff
@@ -1,5 +1,5 @@
 import * as React from 'react';
-import { RenderResult, act } from '@testing-library/react';
+import { RenderResult, act } from '../../../__test__/test-utils';
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/component/Notifications/__test__/FilterNotifications.a11y.test.tsx:2]: as before

File: packages/pn-personafisica-webapp/src/component/Notifications/__test__/FilterNotifications.test.tsx [hunk 35]
```diff
@@ -1,6 +1,6 @@
-import { fireEvent, waitFor, screen, within, RenderResult, act } from '@testing-library/react';
-import * as redux from 'react-redux';
 import { formatToTimezoneString, getNextDay, tenYearsAgo, today } from '@pagopa-pn/pn-commons';
+import * as redux from 'react-redux';
+import { RenderResult, act, fireEvent, screen, waitFor, within } from '../../../__test__/test-utils';
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/component/Notifications/__test__/FilterNotifications.test.tsx:3]: as before

File: packages/pn-personafisica-webapp/src/component/Notifications/__test__/NotificationPayment.a11y.test.tsx [hunk 36]
```diff
@@ -1,15 +1,17 @@
...
 
 jest.mock('react-i18next', () => ({
   // this mock makes sure any components using the translate hook can use it without a warning being shown
   useTranslation: () => ({
-      t: (str: string) => str,
-    }),
-    Trans: () => "mocked-text",
+    t: (str: string) => str,
+  }),
+  Trans: () => 'mocked-text',
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/component/Notifications/__test__/NotificationPayment.a11y.test.tsx:14]: check how is done for the other tests that mocks Trans component

File: packages/pn-personafisica-webapp/src/navigation/__test__/AARGuard.test.tsx [hunk 37]
```diff
@@ -1,13 +1,13 @@
 import React from 'react';
-import { act, screen } from '@testing-library/react';
+import { act, screen } from '../../__test__/test-utils';
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/navigation/__test__/AARGuard.test.tsx:2]: as before

File: packages/pn-personafisica-webapp/src/navigation/__test__/SessionGuard.test.tsx [hunk 38]
```diff
@@ -1,12 +1,12 @@
...
+import { userResponse } from '../../__mocks__/Auth.mock';
+import { act, render, screen } from '../../__test__/test-utils';
 import { authClient } from '../../api/apiClients';
 import { AUTH_TOKEN_EXCHANGE } from '../../api/auth/auth.routes';
-import { userResponse } from '../../redux/auth/__test__/test-users';
-import * as routes from '../routes.const';
 import SessionGuard from '../SessionGuard';
+import * as routes from '../routes.const';
 
 const SessionGuardWithErrorPublisher = () => (
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/navigation/__test__/SessionGuard.test.tsx:11]: check the pa counterpart and remove all the unneeded mocks

File: packages/pn-personafisica-webapp/src/navigation/__test__/ToSGuard.test.tsx [hunk 39]
```diff
@@ -1,7 +1,7 @@
 import { ConsentUser } from "@pagopa-pn/pn-commons";
-import { act, screen } from "@testing-library/react";
 import React from "react";
 import * as redux from "react-redux";
+import { act, screen } from "../../__test__/test-utils";
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/navigation/__test__/ToSGuard.test.tsx:4]: as before

File: packages/pn-personafisica-webapp/src/pages/__test__/Deleghe.page.a11y.test.tsx [hunk 40]
```diff
@@ -1,10 +1,7 @@
...
 import * as redux from 'react-redux';
-
-import * as isMobileHook from '@pagopa-pn/pn-commons/src/hooks/useIsMobile';
-
-import { act, RenderResult } from '@testing-library/react';
-import { axe, render } from '../../__test__/test-utils';
+import { act, axe, render, RenderResult } from '../../__test__/test-utils';
 import Deleghe from '../Deleghe.page';
 
 const useIsMobileSpy = jest.spyOn(isMobileHook, 'useIsMobile');
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/pages/__test__/Deleghe.page.a11y.test.tsx:7]: remove the isMobileSpy and check how it was done in the pa webapp

File: packages/pn-personafisica-webapp/src/pages/__test__/NuovaDelega.test.tsx [hunk 41]
```diff
@@ -1,13 +1,11 @@
-import * as redux from 'react-redux';
-import { fireEvent, waitFor } from '@testing-library/react';
-import * as isMobileHook from '@pagopa-pn/pn-commons/src/hooks/useIsMobile';
 import { RecipientType } from '@pagopa-pn/pn-commons';
-
-import { render } from '../../__test__/test-utils';
-import NuovaDelega from '../NuovaDelega.page';
+import * as isMobileHook from '@pagopa-pn/pn-commons/src/hooks/useIsMobile';
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/pages/__test__/NuovaDelega.test.tsx:2]: Remove the isMobileHook and check how was done in the pa webapp

File: packages/pn-personafisica-webapp/src/pages/__test__/NuovaDelega.test.tsx [hunk 42]
```diff
@@ -1,13 +1,11 @@
...
+import * as redux from 'react-redux';
+import { fireEvent, render, waitFor } from '../../__test__/test-utils';
 import * as actions from '../../redux/newDelegation/actions';
-import * as trackingFunctions from '../../utils/mixpanel';
 import { TrackEventType } from '../../utils/events';
+import * as trackingFunctions from '../../utils/mixpanel';
+import NuovaDelega from '../NuovaDelega.page';
 
 jest.mock('../../component/Deleghe/VerificationCodeComponent', () => ({
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/pages/__test__/NuovaDelega.test.tsx:10]: remove al unneeded mocks

File: packages/pn-personafisica-webapp/src/component/Contacts/__test__/CourtesyContactsList.a11y.test.tsx [hunk 43]
```diff
@@ -14,21 +13,23 @@ jest.mock('react-i18next', () => ({
...
-  code: '12345',
-}, {
-  addressType: 'courtesy',
-  recipientId: 'recipient1',
-  senderId: 'default',
-  channelType: CourtesyChannelType.EMAIL,
-  value: 'test@test.com',
-  code: '54321',
-},
+const mockedContacts: Array<DigitalAddress> = [
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/component/Contacts/__test__/CourtesyContactsList.a11y.test.tsx:16]: use the mocks in the Contacts.mock.ts

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

**Review by AndreaCimini90 — DISMISSED** [review]

**Review by AndreaCimini90 — DISMISSED** [review]

**Review by AndreaCimini90 — APPROVED** [review]

**Pull request merged**